src/cdk/layout/breakpoints-observer.ts

```diff
@@ -75,19 +75,22 @@ export class BreakpointObserver implements OnDestroy {
     const queries = splitQueries(coerceArray(value));
     const observables = queries.map(query => this._registerQuery(query).observable);
 
-    return combineLatest(observables).pipe(
-      debounceTime(0, asapScheduler),
-      map((breakpointStates: InternalBreakpointState[]) => {
-        const response: BreakpointState = {
-          matches: false,
-          breakpoints: {},
-        };
-        breakpointStates.forEach((state: InternalBreakpointState) => {
-          response.matches = response.matches || state.matches;
-          response.breakpoints[state.query] = state.matches;
-        });
-        return response;
-      }));
+    let stateObservable = combineLatest(observables);
+    // Emit the first state immediately, and then debounce the subsequent emissions.
+    stateObservable = concat(
+      stateObservable.pipe(take(1)),
```

Comment:
Can you explain what the intention of `take(1)`, then `skip(1)` here is?  It seems to me that would just cancel each other out.

I think that we might just need to use the default scheduler with `debounceTime`.  After talking a bit with Ben I realized that I was overly aggressive with using the `asapScheduler` because the back to back events in the cases we are seeing are nearly guaranteed to be in different microtasks.

Are you seeing the same issue if you just drop the usage of `asapScheduler` in the code we currently have?

e.g.
```ts
return combineLatest(observables).pipe(
  debounceTime(0),
  map((breakpointStates: InternalBreakpointState[]) => {
    const response: BreakpointState = {
      ...
```

Reply:
Yes, this PR changes the debounce operator to use the default scheduler, AsyncScheduler, because it resolves the issue. The take and skip are required as a result of using the AsyncScheduler which would debounce the first emission until after the view is initialized. To resolve this, take ensures that the first media query is emitted before the components are initialized. Skip skips the query emitted by take. The result is that the first query is emitted immediately and subsequent emissions are debounced.

Reply:
Gotcha, makes sense.  Definitely a clever way to handle it, thanks for the getting this all sorted out for us!

Reply:
You're welcome.